pkg/cleaner/aws/aws.go

```diff
@@ -62,7 +87,33 @@ func (a *Cleaner) Clean() error {
 	return nil
 }
 
-func shouldBeDeleted(stack *cloudformation.Stack) bool {
+func (a *Cleaner) cleanBuckets() error {
+	input := &s3.ListBucketsInput{}
+	output, err := a.s3Client.ListBuckets(input)
+	if err != nil {
+		return err
+	}
+	for _, bucket := range output.Buckets {
+		a.logger.Log("level", "debug", "message", fmt.Sprintf("checking bucket %q", *bucket.Name))
+		if bucketShouldBeDeleted(bucket) {
+			a.logger.Log("level", "debug", "message", fmt.Sprintf("bucket %q should be deleted", *bucket.Name))
+			deleteBucketInput := &s3.DeleteBucketInput{
+				Bucket: bucket.Name,
+			}
+			_, err := a.s3Client.DeleteBucket(deleteBucketInput)
+			if err != nil {
+				// do not return on error, try to continue deleting.
+				a.logger.Log("level", "debug", "message", fmt.Sprintf("could not delete bucket %q: %#v", *bucket.Name, err))
```

Comment:
IMO we should rather retry. I think it is ok to continue after some retries did not succeed either.

Reply:
I disagree here, IMO it's better to keep this code simple and let the cron-like executor do the retries. If something fails in one attempt we are sure that it's going to be retried later, WDYT?

Reply:
I thought it would not be too complicated to use a backoff and retry at max 3 times in case of network or API issues. Do what you think is right.

Reply:
Thanks, going with this.